contributors/design-proposals/storage/volume-topology-scheduling.md

```diff
@@ -266,41 +304,565 @@ const (
 
 This approach allows us to introduce the new binding behavior gradually and to
 be able to maintain backwards compatibility without deprecation of previous
-behavior.  However, it has a few downsides:
+behavior. However, it has a few downsides:
 * StorageClass will be required to get the new binding behavior, even if dynamic
 provisioning is not used (in the case of local storage).
-* We have to maintain two different paths for volume binding.
+* We have to maintain two different code paths for volume binding.
 * We will be depending on the storage admin to correctly configure the
```

Comment:
I don't have any special use case for very slow-to-bind volumes, it would just break existing behavior. Users (and automation) are used to their PVCs being bound relatively quickly and they will be confused.

Someone on SIG-storage said that in some storage backends provisioning can take hours and it's well planned operation, so users want to create PVCs and let them provision before they start a pod. @kangarlou, was that you?

Reply:
@jsafrane No, it wasn't me, but here is my two cents: 
Today for dynamic provisioners, binding happens as soon as the provisioning is done. For the NFS and iSCSI volumes that we manage on NetApp storage, provisioning happens very quickly. To keep it fast, our provisioner doesn't format the iSCSI volumes and relies on kubelet to perform this functionality during attach. This is not a concern for us, but I can see cases where provisioning takes a long time for some plugins/provisioners (e.g., formatting large drives at the time of volume creation, provisioning from snapshots). Having said that, I don't think we should maintain two code paths. As far as external automation is concerned, because provisioning can take arbitrarily long time (in the worst case, the provisioning can fail), any external tooling should inspect the PVC status and not rely on the immediate binding behavior. I agree with @jsafrane that some users prefer to provision their PVCs before they start the pods, but for provisioners that take a long time, I don't think the proposed changes affect the user experience much as such users see provisioning and binding as one step, not as two separate steps.